dpctl/tests/test_usm_ndarray_ctor.py

```diff
@@ -1791,14 +1791,26 @@ def test_full_strides():
     assert np.array_equal(dpt.asnumpy(X), Xnp)
 
 
+def _reduce_to_idtype_range(o, dtype):
+    assert isinstance(dtype, np.dtype)
+    assert dtype.kind in "ui"
+    _iinfo = np.iinfo(dtype)
+    _max = _iinfo.max
+    _min = _iinfo.min
+    return ((o - _min) % (1 + _max - _min)) + _min
+
+
 def test_full_gh_1230():
     q = get_queue_or_skip()
     dtype = "i4"
     dt_maxint = dpt.iinfo(dtype).max
     X = dpt.full(1, dt_maxint + 1, dtype=dtype, sycl_queue=q)
     X_np = dpt.asnumpy(X)
     assert X.dtype == dpt.dtype(dtype)
-    assert np.array_equal(X_np, np.full_like(X_np, dt_maxint + 1))
+    expected = np.full_like(
+        X_np, _reduce_to_idtype_range(dt_maxint + 1, X.dtype)
+    )
+    assert np.array_equal(X_np, expected)
 
     with pytest.raises(OverflowError):
         dpt.full(1, dpt.iinfo(dpt.uint64).max + 1, sycl_queue=q)
```

Comment:
This fix will work, but I would argue this is a temporary fix, we should follow suit with NumPy and disallow Python scalars outside of the dtype range in `full` entirely, which would render this test obsolete.

The issue was originally raised because NumPy allowed this behavior, but we did not.

We can also defer that work to a separate PR.